src/main/java/de/qabel/desktop/DesktopClientGui.java

```diff
@@ -55,63 +54,67 @@ public void start(Stage stage) throws Exception {
-                    runtimeConfiguration.setWindow(((LayoutController) this.view.getPresenter()).getWindow());
-                    Scene layoutScene = new Scene(view, 900, 600, true, aa);
-
-                    Platform.runLater(() -> {
-                        primaryStage.setScene(layoutScene);
-
-                        if (config.hasAccount() && config.getSelectedIdentity() != null) {
-                            primaryStage.setIconified(true);
-                        }
-                    });
-
-                    if (config.getSelectedIdentity() != null) {
-                        addShareMessageRenderer(config.getSelectedIdentity());
-                    }
-                } catch (Exception e) {
-                    logger.error("failed to init background services: " + e.getMessage(), e);
-                    Platform.runLater(() -> {
-                        final CrashReportAlert alert = new CrashReportAlert(
-                            services.getCrashReportHandler(),
-                            "failed to init brackground services",
-                            e
-                        );
-                        alert.showAndWait();
-                        System.exit(-1);
-                    });
-                }
-            })
+            initBackgroundServices()
         );
         config.onSelectIdentity(this::addShareMessageRenderer);
+        primaryStage.show();
 
-        QabelTray tray = new QabelTray(primaryStage, new AwtToast());
-        tray.install();
-        Runtime.getRuntime().addShutdownHook(new Thread() {
-            @Override
-            public void run() {
-                Platform.exit();
+        QabelTray tray = getQabelTray();
+        trayNotifications(tray);
+
+        if (config.hasAccount()) {
+            primaryStage.close();
+            System.out.println(" primaryStage closed ");
+        }
+    }
+
+    @NotNull
+    private Future<?> initBackgroundServices() {
+        return executorService.submit(() -> {
+            try {
+                startTransferManager();
+                startSyncDaemon();
+                startDropDaemon();
+
+                loggedInStage();
+                if (config.getSelectedIdentity() != null) {
+                    addShareMessageRenderer(config.getSelectedIdentity());
+                }
+            } catch (Exception e) {
+                logger.error("failed to init background services: " + e.getMessage(), e);
+                Platform.runLater(() -> {
+                    final CrashReportAlert alert = new CrashReportAlert(
+                        services.getCrashReportHandler(),
+                        "failed to init brackground services",
+                        e
+                    );
+                    alert.showAndWait();
+                    System.exit(-1);
+                });
             }
         });
-        primaryStage.show();
+    }
+
+    private void loggedInStage() {
```

Comment:
try to show actions with your method names. `loggedInStage` could either be `getLoginStage` or `toLoginStage` or `showLoginStage`. Not THAT important on an internal method but it increases the readability of your code